karapace/schema_reader.py

```diff
@@ -362,25 +371,17 @@ def _handle_msg_schema(self, key: dict, value: Optional[dict]) -> None:
         with self.id_lock:
             # dedup schemas to reduce memory pressure
             schema_str = self._hash_to_schema.setdefault(hash(schema_str), schema_str)
-
-            typed_schema = TypedSchema(
-                schema_type=schema_type_parsed,
-                schema_str=schema_str,
-            )
-            schema = {
-                "schema": typed_schema,
-                "version": schema_version,
-                "id": schema_id,
-                "deleted": schema_deleted,
-            }
+            self._hash_to_schema_id_on_subject.setdefault(hash((schema_subject, schema_str)), SchemaId(schema_id))
```

Comment:
Inside this `with` statement, beginning has this line setting value into hashmap, but if it was actually `schema_deleted=True`, then this same(?) is removed some lines later?

Also, for get & delete there is function to do those, but not for setting.

Reply:
I'll add the function.